### PR TITLE
QOL-8595 move shared features from Data extension to ckanext-qgov

### PR DESCRIPTION
**Pull request by ThrawnCA**

### CODE REVIEW

**Review by duttonw — APPROVED** [review]

**Review by duttonw — APPROVED** [review]

File: test/features/resources.feature [hunk 1]
```diff
@@ -20,23 +20,3 @@ Feature: Resource UI
         When I create a resource with name "Non-HTTP link" and URL "git+https://github.com/ckan/ckan.git"
         And I press the element with xpath "//a[contains(@title, 'Non-HTTP link') and contains(string(), 'Non-HTTP link')]"
         And I should see "http://git+https://github.com/ckan/ckan.git"
-
-    Scenario Outline: Link resource with private address should be rejected
```
[the remark this answers is not in the document]
Comment (ThrawnCA) [reply test/features/resources.feature:24]: The whole feature was never a reliable approach, added a bunch of overhead (DNS lookups), and is superseded by our Squid-based filter.

File: ckanext/qgov/common/user_creation/commands.py [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+    summary = __doc__.split('\n')[0]
+    usage = __doc__
+    max_args = 0
+    min_args = 0
+
+    def command(self):
+        self._load_config()
+
+        # Below is 18th May 2021 10.42am in UTC.
+        date_threshold = datetime(2021, 5, 18, 00, 42)
```
Comment (duttonw) [inline ckanext/qgov/common/user_creation/commands.py:22]: I wish they set this as a variable..

File: ckanext/qgov/common/user_creation/commands.py [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+
+        model.Session.query(model.User) \
+            .filter(sa.not_(sa.func.lower(sa.func.coalesce(model.User.fullname, '')).like('data.qld%'))) \
+            .filter(model.User.created < date_threshold) \
+            .update({'fullname': 'Displayed name required'}, synchronize_session='fetch')
```
Comment (duttonw) [inline ckanext/qgov/common/user_creation/commands.py:27]: fullname should be a has of the username so they are still unique when looking..

File: ckanext/qgov/common/user_creation/validators.py [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+        return toolkit.g.userobj
+    else:
+        return None
+
+
+def data_qld_user_name_validator(key, data, errors, context):
+    user = _get_user()
+    is_sysadmin = user is not None and user.sysadmin
+
+    if not is_sysadmin and 'publisher' in data[key].lower():
```
Comment (duttonw) [inline ckanext/qgov/common/user_creation/validators.py:18]: are we going to fix salsa's mess. ie. check context and see if they are indeed altering their username/display name?

File: test/features/users.feature [hunk 5]
```diff
@@ -137,13 +137,9 @@ Feature: User APIs
...
         Then I should see an element with xpath "//h2[contains(string(), 'News feed')]"
 
-    Scenario: Dashboard page is not accessible anonymously
-        When I go to the dashboard
-        Then I should see an element with xpath "//*[contains(string(), 'Not authorised to see this page')]"
-
 
     Scenario: Password reset works
-        When I request a password reset for "publisher"
+        When I request a password reset for "editor"
```
Comment (duttonw) [inline test/features/users.feature:142]: can you add a test for test change password with special key word block based on user_creation.feature 

can you also add tag's for these tests which are related to the username/display name validation.

**Review by duttonw — COMMENTED** [review]
looking ok. btw is ckan-ex-qgov shared between data and publications or is it data only?

**Review by duttonw — APPROVED** [review]

File: ckanext/qgov/common/user_creation/validators.py (hunk 4 above)
Comment (ThrawnCA) [reply ckanext/qgov/common/user_creation/validators.py:18]: It would be a good thing to add, yes.

File: test/features/users.feature (hunk 5 above)
Comment (ThrawnCA) [reply test/features/users.feature:142]: Can do.

File: ckanext/qgov/common/user_creation/validators.py (hunk 4 above)
Comment (ThrawnCA) [reply ckanext/qgov/common/user_creation/validators.py]: This is done.

File: test/features/users.feature (hunk 5 above)
Comment (ThrawnCA) [reply test/features/users.feature:142]: The relevant tests are already in `user_creation.feature` and are tagged `@user_creation`.

**Comment by ThrawnCA** [comment]
> looking ok. btw is ckan-ex-qgov shared between data and publications or is it data only?

It's shared; that's its purpose.

**Pull request merged**